File: app/src/main/java/app/attestation/auditor/SystemProperties.java
package app.attestation.auditor;

import java.io.IOException;
import java.util.Scanner;

class SystemProperties {
    public static String get(final String key, final String def) {
        try {
            final Process process = new ProcessBuilder("getprop", key, def).start();
            try (Scanner scanner = new Scanner(process.getInputStream())) {
                return scanner.nextLine().trim();
            }
        } catch (IOException e) {}
        return def;
    }
}

File: app/src/main/java/app/attestation/auditor/SystemProperties.kt
package app.attestation.auditor

import java.io.IOException
import java.util.Scanner

object SystemProperties {
    @JvmStatic
    fun get(key: String, def: String): String {
        try {
            val process = ProcessBuilder("getprop", key, def).start()
            Scanner(process.inputStream).use {
                return it.nextLine().trim()
            }
        } catch (_: IOException) {
            return def
        }
    }
}
